ts/Core/Axis/OrdinalAxis.ts

```diff
@@ -858,84 +981,76 @@ namespace OrdinalAxis {
+                    ordinal.extendedOrdinalPositions = extendedOrdinalPositions;
+                }
+
+                if (extendedOrdinalPositions && extendedOrdinalPositions.length) {
+                    const indexInEOP = ordinal.getIndexOfPoint(val, extendedOrdinalPositions),
+                        mantissa = correctFloat(indexInEOP % 1);
 
-                let ordinalSlope = ordinal.slope,
-                    ordinalOffset = ordinal.offset,
-                    i = ordinalPositions.length - 1,
-                    linearEquivalentLeft,
-                    linearEquivalentRight,
-                    distance;
-
-
-                // Handle the case where we translate from the index directly,
-                // used only when panning an ordinal axis
-                if (fromIndex) {
-
-                    if (val < 0) { // out of range, in effect panning to the left
-                        val = ordinalPositions[0];
-                    } else if (val > i) { // out of range, panning to the right
-                        val = ordinalPositions[i];
-                    } else { // split it up
-                        i = Math.floor(val);
-                        distance = val - i; // the decimal
+                    // Check if the index is inside extendedOrdinalPositions.
+                    // If true, read/approximate value for that exact index.
+                    if (indexInEOP >= 0 && indexInEOP < extendedOrdinalPositions.length) {
+                        const leftNeighbour = extendedOrdinalPositions[Math.floor(indexInEOP)],
+                            rightNeighbour = extendedOrdinalPositions[Math.ceil(indexInEOP)],
+                            distance = rightNeighbour - leftNeighbour;
+
+                        return extendedOrdinalPositions[Math.floor(indexInEOP)] + mantissa * distance;
                     }
 
-                // Loop down along the ordinal positions. When the linear
-                // equivalent of i matches an ordinal position, interpolate
-                // between the left and right values.
-                } else {
-                    while (i--) {
-                        linearEquivalentLeft =
-                            ((ordinalSlope as any) * i) + (ordinalOffset as any);
-                        if (val >= linearEquivalentLeft) {
-                            linearEquivalentRight =
-                                ((ordinalSlope as any) *
-                                (i + 1)) +
-                                (ordinalOffset as any);
-                            // something between 0 and 1
-                            distance = (val - linearEquivalentLeft) /
-                                (linearEquivalentRight - linearEquivalentLeft);
-                            break;
-                        }
+                    const EOPlength = extendedOrdinalPositions.length,
+                        leftVisiblePoint = extendedOrdinalPositions[0],
```

Comment:
since You are taking the first and last index of extendedOrdinalPositions, the variable name is a bit misleading, because this point doesn't have to be visible (mostlikely isn't)